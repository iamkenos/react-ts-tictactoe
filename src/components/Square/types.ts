export interface SquareProps {
  value: string;
}
