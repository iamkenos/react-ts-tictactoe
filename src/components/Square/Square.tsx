import React from "react";
import "./styles.css";

class Square extends React.Component {
  render() {
    return <button className="square">X</button>;
  }
}

export default Square;
