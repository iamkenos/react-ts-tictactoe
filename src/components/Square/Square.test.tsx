import { render } from "@testing-library/react";
import Square from "./Square";

describe("Square Component", () => {
 let fragment: DocumentFragment;
  it("should render the square component", () => {
    givenRenderSquareComponent();
    thenSquareComponentSnapshotMatch();
  });

  function givenRenderSquareComponent() {
    const { asFragment } = render(<Square value={""}/>);
    fragment = asFragment();
  }

  function thenSquareComponentSnapshotMatch() {
    expect(fragment).toMatchSnapshot();
  }
});
