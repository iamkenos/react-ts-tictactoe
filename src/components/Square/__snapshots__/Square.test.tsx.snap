// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Component should render the square component 1`] = `
<DocumentFragment>
  <button
    class="square"
  >
    X
  </button>
</DocumentFragment>
`;
